src/app/api/signup/route.ts

```diff
@@ -35,60 +32,51 @@ export const POST = withAppRouterHighlight(async (request: NextRequest) => {
     });
   }
 
-  try {
-    const userId = generateDatabaseId("user");
-    const hashedPassword = await new Argon2id().hash(password);
+  const userId = generateDatabaseId("user");
```

Comment:
this should actually be in a try catch, and make sure that it always return the same error, such that you cannot distinguish from "account already exits" and "sign up failed". If the account already exists, you can send an email to that address saying you already have an account.